language/yield_spec.rb

```diff
@@ -211,5 +211,13 @@ class << Object.new
 
       -> { eval(code) }.should raise_error(SyntaxError, /Invalid yield/)
     end
+
+    it 'raises a SyntaxError in top-level outside of the method definition' do
+      code = <<~RUBY
+        1.times { yield }
+      RUBY
+
+      -> { eval(code) }.should raise_error(SyntaxError, /Invalid yield/)
+    end
```

Comment:
I suppose the new test shouldn't be nested into the `describe "Using yield in a singleton class literal" do` section.

Reply:
It seems the new behavior changed in Ruby 2.7, not 3.0.

In Ruby 2.6 such yield leads to `no block given (yield) (LocalJumpError)`  but in Ruby 2.7 - `Invalid yield`.

Reply:
It is:

```ruby
describe "Using yield in a singleton class literal" do
  ruby_version_is ""..."3.0" do
    # ...
  end

  ruby_version_is "3.0" do
    it 'raises a SyntaxError in singleton class definition' do
      # ...
    end

    it 'raises a SyntaxError in top-level outside of the method definition' do
      code = <<~RUBY
        1.times { yield }
      RUBY

      -> { eval(code) }.should raise_error(SyntaxError, /Invalid yield/)
    end
  end
end

```

Reply:
Yes, but it *shouldn't* be, because this example is not a "yield in a singleton class literal".
Could you move it outside, in a separate `describe`?

Also note it's not really top-level here with eval, it's inside 2-3-4 non-lambda blocks, but I think we can gloss over that.

Reply:
@andrykonchin Yes, you are right about the version, I wasn't correctly testing and was confused by `chruby`, my fault.

1. I'm going to change condition for version guard to be 2.7 and up
2. I'm going to pull it outside and give it a better name (will try my best)

Reply:
Since we support everything 2.7 and up I will omit the version guard. Fixed